bpf/bpf_overlay.c

```diff
@@ -50,18 +50,27 @@ static __always_inline int handle_ipv6(struct __ctx_buff *ctx,
 
 	decrypted = ((ctx->mark & MARK_MAGIC_HOST_MASK) == MARK_MAGIC_DECRYPT);
 	if (decrypted) {
-		*identity = get_identity(ctx);
+		*identity = key.tunnel_id = get_identity(ctx);
```

Comment:
@jrfastab the previous version of this was ignored in any case other than when the packet was dropped. Was this intentional? Are you concerned about any potential encryption breakage by this commit?